src/sql/src/optimizer_metrics.rs

```diff
@@ -85,7 +85,8 @@ impl OptimizerMetrics {
                 transform_times = serde_json::to_string(&transform_times)
                     .unwrap_or_else(|_| format!("{:?}", transform_times)),
                 duration = format!("{}ms", duration.as_millis()),
-                "optimizer took more than 500ms"
+                "optimizer took more than {}ms",
+                self.e2e_optimization_time_seconds_log_threshold.as_millis(),
```

Comment:
Oh, maybe extract it from the format string:
```suggestion
                threshold = self.e2e_optimization_time_seconds_log_threshold.as_millis(),
                "slow optimization",

```

Reply:
Ok, done